### ISSUE TITLE
[java] UnusedPrivateMethod when 'var' is used with ternary operator

### ISSUE THREAD

**Issue by kdebski85** [pmd/pmd#5113]
**Affects PMD Version:** 
7.3.0

**Rule:**

[UnusedPrivateMethod](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**

`UnusedPrivateMethod` is reported when a private method is used in an argument of another method called on a reference o type `var` and when a ternary operator was used to assign value to this reference. 

There is no FP when `var` is changed to `Optional<Boolean>`.
It does not matter if method reference to `dummy` is used or  `dummy`  is called in lambda.
Tested with Java 21.

**Code Sample demonstrating the issue:**

```java
import java.util.Optional;

public class VarTest {
    public Optional<Boolean> foo(int param) {
        var optional = param == 0 ? Optional.of(true) : Optional.of(false);
        return optional.flatMap(this::dummy);
    }

    private Optional<Boolean> dummy(boolean foo) {
        return Optional.of(foo);
    }
}

```

**Expected outcome:**

PMD reports a violation at line 8, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by oowekyala** [comment]
After noticing that this is not reproducible on master I found that this was fixed in PMD 7.5.0 by #5176.
